modules/uhsg_active_degree_programme/src/PageCache/DenyActiveDegreeProgramme.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Drupal\uhsg_active_degree_programme\PageCache;
+
+use Drupal\Core\PageCache\ResponsePolicyInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class DenyActiveDegreeProgramme implements ResponsePolicyInterface {
+
+  /**
+   * @inheritdoc
+   */
+  public function check(Response $response, Request $request) {
```

Comment:
This will kill the page cache for majority of the users (since browsing the site having a degree programme selected is a common use case) and cause severe performance issues. Need to find something less aggressive.